### ISSUE TITLE
[ISSUE] Issue with `databricks_group_member` resource

### ISSUE THREAD

**Issue by nihil0** [databricks/terraform-provider-databricks#2071]
### Configuration
```hcl
data "databricks_service_principal" "kadp_sp_dev" {

    provider = databricks.account
    
    application_id = local.kadp_sp_id_dev
}

resource "databricks_group_member" "add_kadp_sp_dev" {

    provider = databricks.account

    group_id = module.adp_admins_group_dev.id
    member_id = data.databricks_service_principal.kadp_sp_dev.id
}
```

### Expected Behavior

The service principal `kadp_sp_dev` should be added as a member of the `adp_admins_group_dev` 

### Actual Behavior

Error message 
```
Error: cannot read group member: Group has no member
│ 
│   with databricks_group_member.add_kadp_sp_dev,
│   on kadp_resources.tf line 32, in resource "databricks_group_member" "add_kadp_sp_dev":
│   32: resource "databricks_group_member" "add_kadp_sp_dev" {
```

### Steps to Reproduce

`terraform plan` succeeds with the output:
```
# databricks_group_member.add_kadp_sp_dev will be created
  + resource "databricks_group_member" "add_kadp_sp_dev" {
      + group_id  = "346046684757246"
      + id        = (known after apply)
      + member_id = "_"
    }
```

### Terraform and provider versions

```
erraform v1.3.8
on darwin_arm64
+ provider registry.terraform.io/databricks/databricks v1.2.1
+ provider registry.terraform.io/hashicorp/aws v4.29.0
```

### Debug Output

```
provider.terraform-provider-databricks_v1.2.1: Response contains error diagnostic: @module=sdk.proto diagnostic_detail= diagnostic_summary="cannot read group member: Group has no member" tf_proto_version=5.3 tf_provider_addr=provider @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:55 tf_rpc=ApplyResourceChange tf_req_id=3747cf9c-4a9f-adbf-2879-ccd1056ce1b6 diagnostic_severity=ERROR tf_resource_type=databricks_group_member timestamp=2023-03-06T15:45:26.492+0200
2023-03-06T15:45:26.496+0200 [ERROR] vertex "databricks_group_member.add_kadp_sp_dev" error: cannot read group member: Group has no member
2023-03-06T15:45:26.498+0200 [DEBUG] states/remote: state read serial is: 77; serial is: 77
2023-03-06T15:45:26.498+0200 [DEBUG] states/remote: state read lineage is: 7d4141e2-76f0-63c1-4097-8102031498d2; lineage is: 7d4141e2-76f0-63c1-4097-8102031498d2
╷
│ Error: cannot read group member: Group has no member
│ 
│   with databricks_group_member.add_kadp_sp_dev,
│   on kadp_resources.tf line 32, in resource "databricks_group_member" "add_kadp_sp_dev":
│   32: resource "databricks_group_member" "add_kadp_sp_dev" {
│ 
```

**Comment by nihil0** [comment]
Error seems to originate from here: https://github.com/databricks/terraform-provider-databricks/blob/master/scim/resource_group_member.go#L23

**Comment by nihil0** [comment]
After some troubleshooting, I realized that the example in the docs [here](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/service_principal) is wrong. It should be 

```hcl
resource "databricks_group_member" "i-am-admin" {
  group_id  = data.databricks_group.admins.id
  member_id = databricks_service_principal.sp.id
}
```

**Comment by nfx** [comment]
possible relevant one https://github.com/databricks/terraform-provider-databricks/issues/2129

**Comment by nkvuong** [comment]
Fixed in #1885
